### PR TITLE
Destroy spectrum on jQuery $destroy event instead of $scope.$destroy event

### LINKED ISSUE #30: Destroy spectrum on jQuery $destroy event instead of $scope.$destroy event
There is a time gap between broadcasting $scope.$destroy and removing the part of DOM. This causes that the input with hex value is displayed instead of colorpicker during $scope destruction

According to the angular documentation (https://code.angularjs.org/1.2.27/docs/api/ng/type/$rootScope.Scope)  there is also jQuery $destroy event emitted:

"Note that, in AngularJS, there is also a $destroy jQuery event, which can be used to clean up DOM bindings before an element is removed from the DOM."

Binding to this event causes that the spectrum is removed smoothly - I've tried it already:

```javascript
$element.on('$destroy', function(){
        	  $input.spectrum('destroy');
});
```

I can submit a PR, if this change is accepted.

### PR DESCRIPTION
**Pull request by tomaszkrym**
Resolves #30

### CODE REVIEW

File: test/unit/spectrumDirectiveSpec.js [hunk 1]
```diff
@@ -137,7 +137,7 @@ describe('SpectrumDirective', function() {
         'options': 'options'
       });
       expect($('.sp-container').length).toBe(1);
-      d.scope.$destroy();
```
Comment (joscha) [inline test/unit/spectrumDirectiveSpec.js]: don't we have to emit both here? otherwise the `.off()` is never called?

**Comment by joscha** [comment]
Thanks @tomaszkrym - would you be able to rebase the PR so that it does not contain a changed dist file and add a test that shows the shortcomings of the old approach?

File: test/unit/spectrumDirectiveSpec.js (hunk 1 above)
Comment (tomaszkrym) [inline test/unit/spectrumDirectiveSpec.js]: The off() call is checked in test 'should deregister the handler on directive destroy'. I left the 'off()' call in $scope.$destroy listener, as IMO the click should be disabled as soon as it is known that the directive is about to be destroyed.

**Comment by tomaszkrym** [comment]
@joscha the dist file is not in .gitignore, so this is why I commited it. Furthermore, if anybody had a bower dependency to branch master, it would have installed dist files that were out of sync with src files.

It is hardly possible to show the shortcomings of the old approach (blinking input during destroy) in the unit test - the visiblity is updated asynchronously.

**Pull request merged**